### PR TITLE
PCRM-5183 dps filename change

### PR DESCRIPTION
**Pull request by johnwaghorn**

### CODE REVIEW

File: test/unit/validate_and_forward/test_validate_and_forward.py [hunk 1]
```diff
@@ -44,40 +44,40 @@ def test_validate_and_forward_valid_file_is_successfully_sent_to_dps_outbound_bu
...
     def test_validate_and_forward_invalid_file_is_rejected_and_not_sent_to_dps_outbound_bucket(
         self,
         lambda_handler,
         lambda_context,
         invalid_file_event,
         create_rejection_bucket,
+        create_dps_outbound_bucket,
         create_dynamodb_table,
     ):
-        app = lambda_handler
```
Comment (antonyibbotson1-nhs) [inline test/unit/validate_and_forward/test_validate_and_forward.py:73]: Can you remove this in all the other tests to keep it consistent please.

File: test/unit/validate_and_forward/test_validate_and_forward.py [hunk 2]
```diff
@@ -44,40 +44,40 @@ def test_validate_and_forward_valid_file_is_successfully_sent_to_dps_outbound_bu
...
         expected_message = "Lambda application stopped gp extract input file rejected, rejected file handled"
         expected_file = INVALID_FILE
 
-        response = app.main(invalid_file_event, lambda_context)
+        response = lambda_handler.main(invalid_file_event, lambda_context)
```
Comment (antonyibbotson1-nhs) [inline test/unit/validate_and_forward/test_validate_and_forward.py:75]: Same as other comment, make this change in all tests to keep it consistent

Comment (johnwaghorn) [reply test/unit/validate_and_forward/test_validate_and_forward.py:75]: will do for the rest

File: test/unit/validate_and_forward/conftest.py [hunk 3]
```diff
@@ -84,7 +85,10 @@ def get_body_from_file_in_s3(bucket_name: str) -> str:
     for s3_object in bucket_objects.get("Contents", []):
         object_key = s3_object["Key"]
     file = client.get_object(Bucket=bucket_name, Key=str(object_key))
-    return file["Body"].read().decode("utf-8")
+    try:
+        return file["Body"].read().decode("utf-8")
+    except UnicodeDecodeError:
+        return file["Body"].read().decode("latin1")
```
Comment (antonyibbotson1-nhs) [inline test/unit/validate_and_forward/conftest.py:91]: This kinda feels like a pointless addition at the moment :thinking:

File: packages/validate_and_forward/lambda_handler.py [hunk 4]
```diff
@@ -51,7 +51,7 @@ def start(self):
         # get gp_file extract stats for job's table
         # Also handles rejection if file can't be parsed and validations fails
         try:
-            practice_code, num_of_records = self.parse_gp_extract(
+            self.practice_code, num_of_records = self.parse_gp_extract(
```
Comment (antonyibbotson1-nhs) [inline packages/validate_and_forward/lambda_handler.py:54]: why?

File: packages/validate_and_forward/lambda_handler.py [hunk 5]
```diff
@@ -173,10 +173,13 @@ def get_file_contents(self):
             file = body.decode("latin1")
         return filename, file
 
-    def send_file_to_dps_outbound(self, filename: str, file):
+    def send_file_to_dps_outbound(self, file):
```
Comment (antonyibbotson1-nhs) [inline packages/validate_and_forward/lambda_handler.py:176]: type annotation?

File: packages/validate_and_forward/lambda_handler.py [hunk 6]
```diff
@@ -173,10 +173,13 @@ def get_file_contents(self):
...
+
         try:
             self.s3.put_object(
-                Bucket=self.DPS_OUTBOUND, Key=f"{filename}-{self.job_id}", Body=file
+                Bucket=self.DPS_OUTBOUND,
+                Key=f'{datetime.now().strftime("%Y%m%d-%H%M%S")}-{self.practice_code}-{self.job_id}.csv',
```
Comment (antonyibbotson1-nhs) [inline packages/validate_and_forward/lambda_handler.py:181]: Would be tempted to break this out into a separate var, it's a bit awkward

File: test/unit/validate_and_forward/conftest.py (hunk 3 above)
Comment (johnwaghorn) [reply test/unit/validate_and_forward/conftest.py:91]: I wouldn't say it's pointless. Latin1 encoded data exists in the unit tests for this Lambda.

Comment (antonyibbotson1-nhs) [reply test/unit/validate_and_forward/conftest.py:91]: Where is it being used with this method?

File: packages/validate_and_forward/lambda_handler.py (hunk 4 above)
Comment (johnwaghorn) [reply packages/validate_and_forward/lambda_handler.py:54]: for the filename

**Review by antonyibbotson1-nhs — APPROVED** [review]

**Pull request merged**